### ISSUE TITLE
Question, how does it work?

### ISSUE THREAD

**Issue by spyhunter99** [skywinder/github-changelog-generator#486]
Just a question out of curiosity (and because I'm making a maven plugin that does something similar).

Q: how do you go about calculating which ticket was resolved with a given version?

I would imagine that are there several different ways to go about this.

 - time stamp, find all tickets resolved between the release tags
 - milestone based? not all projects use milestones but i would imagine you'd have a fallback plan
 - git commit log base? are you basing the resolve issues based on the git commit log 

Q: how do you determine which tags are actually releases (or releases as tags)?

It's possible to have a tag without a release, but not a release without a tag. Are you pulling releases from the github api?

**Comment by olleolleolle** [comment]
Hello! 

Explore the `--help` for the options for what's possible. The code also is available to read, here.

The `@tag_section_mapping` calculation ought to interest you:
https://github.com/skywinder/github-changelog-generator/blob/bcacabcf5fca8eeaba7624b7f5551c9d92244b41/lib/github_changelog_generator/generator/generator_tags.rb#L16

The _GitHub Releases_ feature is not used. It's all about Git tags and time-ranges and GitHub PRs.

A counter-question: Would you expect to find an overview of _How this works_ in the README document?

**Comment by spyhunter99** [comment]
THanks for the reply, I'm an experienced java developer but ruby is like greek to me. I would have normally traced through the code otherwise (and I attempted it).

As for the readme section, sounds like a great idea!

In my side project, I was able to pull the "release" lists from github and then calculate time stamps from that. But then I started thinking about tags. Tags could be on any branch, not necessarily on master and tags may not necessarily represent release versions. Also issues could be marked closed at a given date but may not necessarily be merged for a given release or tag. I guess there's no way to predict what ways someone could use git and the github infrastructure. My head hurts thinking about it.

I'll post the code shortly and link it if you are interested but this project was the inspiration.
